config/routes.rb

```diff
@@ -64,6 +64,7 @@
 
     resources :application_users, only: [:index] do
       collection do
+        get 'username/:username', action: 'username'
```

Comment:
That seems like an odd route to me... it looks more like you want 3 different routes.